snuba/state/explain_meta.py

```diff
@@ -27,7 +31,7 @@ def add_step(category: str, name: str, data: dict[str, Any] | None = None) -> No
     try:
         if data is None:
             data = {}
-        step = ExplainStep(category, name, data)
+        step = ExplainStep(category, "query_transform", name, data)
```

Comment:
What is this type used for if it's always defaulted to `query_transform`? In the future, what other types would you imagine?

Reply:
Good question! I have a couple other types I thought of:
- validation steps, that record information about why a query isn't valid
- splitter steps, that show how the query was split (if at all)
- caching step, just shows whether the query was cached, waited etc.
- allocation step, would this query have been throttled? Why?
- rate limit steps, would this query have been rate limited? Why?